contracts/sg-ics721/src/execute.rs

```diff
@@ -0,0 +1,75 @@
+use cosmwasm_std::{from_binary, to_binary, Addr, Binary, DepsMut, Env, StdResult};
+use ics721::{
+    execute::Ics721Execute,
+    state::CollectionData,
+    token_types::Class,
+    utils::{convert_owner_chain_address, get_collection_data},
+};
+use sg721_base::msg::{CollectionInfoResponse, QueryMsg};
+
+use crate::state::{SgCollectionData, SgIcs721Contract};
+
+impl Ics721Execute for SgIcs721Contract {
+    type ClassData = SgCollectionData;
+
+    /// sg-ics721 sends custom SgCollectionData, basically it extends ics721-base::state::CollectionData with additional collection_info.
+    fn get_class_data(&self, deps: &DepsMut, sender: &Addr) -> StdResult<Self::ClassData> {
+        let CollectionData {
+            owner,
+            contract_info,
+            name,
+            symbol,
+            num_tokens,
+        } = get_collection_data(deps, sender)?;
+        let collection_info: CollectionInfoResponse = deps
+            .querier
+            .query_wasm_smart(sender, &QueryMsg::CollectionInfo {})?;
+
+        Ok(SgCollectionData {
+            owner,
+            contract_info,
+            name,
+            symbol,
+            num_tokens,
+            collection_info,
+        })
+    }
+
+    fn init_msg(&self, env: &Env, class: &Class) -> StdResult<Binary> {
+        let creator = match class.data.clone() {
+            // in case no class data is provided, ics721 will be used as the creator.
+            None => env.contract.address.to_string(),
```

Comment:
@humanalgorithm, @shanev what do we do, incase there is no owner in ClassData provided? This happens e.g. when collection comes from nft-transfer module (IRISnet, OmniFlix) or some other custom ICS721 contract! Rn it uses sg-ics721 contract address being the creator.
One possibility is using owner address.

Reply:
@humanalgorithm @shanev, I changed this. Creator is now not ics721 contract itself, but it's creator (of ICS721 contract). ICS721 is instantiated with a multisig. This will be also the plan for mainnet.